ouroboros-consensus/test-consensus/Test/Consensus/Ledger/Byron.hs

```diff
@@ -0,0 +1,186 @@
+{-# LANGUAGE FlexibleInstances   #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Test.Consensus.Ledger.Byron (tests) where
+
+import           Codec.CBOR.Decoding (Decoder)
+import           Codec.CBOR.Encoding (Encoding)
+import           Codec.CBOR.Read (deserialiseFromBytes)
+import           Codec.CBOR.Write (toLazyByteString)
+import qualified Data.ByteString as Strict
+import qualified Data.ByteString.Lazy as Lazy
+
+import           Cardano.Binary (ByteSpan, slice)
+import           Cardano.Chain.Block (ABlockOrBoundary (..))
+import qualified Cardano.Chain.Block as CC.Block
+import           Cardano.Chain.Slotting (EpochSlots (..))
+import           Cardano.Chain.UTxO (UTxOValidationError, annotateTxAux)
+import           Cardano.Crypto (ProtocolMagicId (..))
+
+import           Ouroboros.Network.Block (HeaderHash)
+
+import           Ouroboros.Consensus.Block (Header)
+import           Ouroboros.Consensus.Ledger.Byron
+import           Ouroboros.Consensus.Ledger.Byron.Config
+import           Ouroboros.Consensus.Mempool.API (ApplyTxErr)
+
+import           Test.QuickCheck
+import           Test.QuickCheck.Hedgehog (hedgehog)
+import           Test.Tasty
+import           Test.Tasty.QuickCheck
+
+import qualified Test.Cardano.Chain.Block.Gen as CC.Gen
+import qualified Test.Cardano.Chain.UTxO.Gen as CC.Gen
+
+
+tests :: TestTree
+tests = testGroup "Byron"
+  [ testGroup "Serialisation roundtrips"
+      [ testProperty "roundtrip Block"       prop_roundtrip_Block
```

Comment:
Let's make sure that this is exhaustive, and if not, mark the remaining elements as TODO. Might be useful to add a comment to the Byron to say if we add more encoders/decoders, they should have corresponding tests here.